src/toxcore/network.rs

```diff
@@ -22,29 +22,205 @@
 //! Networking part of the toxcore.
 
 
-use std::net::UdpSocket;
+use std::any::Any;
+use std::rc::Rc;
+use std::cell::RefCell;
+use std::io::{ self, ErrorKind };
+use std::sync::{ Once, ONCE_INIT };
+use std::ops::{ Range, RangeFrom, RangeTo, RangeFull };
+use std::net::{ IpAddr, Ipv6Addr, SocketAddr, ToSocketAddrs };
+use std::collections::HashMap;
+use mio::udp::UdpSocket;
+use sodiumoxide;
 
 
 /// Minimum port which Tox will try to bind to.
 pub const PORT_MIN: u16 = 33445;
 /// Maximum port which Tox will try to bind to.
 pub const PORT_MAX: u16 = 33545;
+/// Maximum size of a UDP packet that tox will handle.
+pub const MAX_UDP_PACKET_SIZE: usize = 2048;
+
+static AT_STARTUP_RAN: Once = ONCE_INIT;
+
+/** Run this before creating sockets.
+
+    return true on success
+    return false on failure
+*/
+fn networking_at_startup() -> bool {
+    let mut output = true;
+    AT_STARTUP_RAN.call_once(|| output = sodiumoxide::init());
+    output
+}
```

Comment:
https://github.com/irungentoo/toxcore/blob/master/toxcore/network.c#L439

I also think that it should not be here, but it should be placed where it?